src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,225 @@
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <data/DBHelpers.h>
+#include <rpc/AMMHelpers.h>
+#include <rpc/handlers/AMMInfo.h>
+
+#include <ripple/protocol/AMMCore.h>
+
+namespace {
+
+std::string
+toIso8601(ripple::NetClock::time_point tp)
+{
+    using namespace std::chrono;
+    static auto constexpr rippleEpochOffset = seconds{rippleEpochStart};
+
+    return date::format(
+        "%Y-%Om-%dT%H:%M:%OS%z",
+        date::sys_time<system_clock::duration>(system_clock::time_point{tp.time_since_epoch() + rippleEpochOffset})
+    );
+};
+
+}  // namespace
+
+namespace rpc {
+AMMInfoHandler::Result
+AMMInfoHandler::process(AMMInfoHandler::Input input, Context const& ctx) const
+{
+    using namespace ripple;
+
+    if (!input.ammAccount && (input.issue1 == ripple::noIssue() || input.issue2 == ripple::noIssue()))
+        return Error{Status{RippledError::rpcINVALID_PARAMS}};
+
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence
+    );
+
+    if (auto const status = std::get_if<Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<LedgerInfo>(lgrInfoOrStatus);
+    ripple::uint256 ammID;
+    if (input.ammAccount) {
+        auto accountKeylet = ripple::keylet::account(*input.ammAccount);
+        auto const accountLedgerObject =
+            sharedPtrBackend_->fetchLedgerObject(accountKeylet.key, lgrInfo.seq, ctx.yield);
+        ripple::STLedgerEntry const sle{
+            ripple::SerialIter{accountLedgerObject->data(), accountLedgerObject->size()}, accountKeylet.key};
+        ammID = sle.getFieldH256(ripple::sfAMMID);
+    }
+
+    if (input.accountID) {
+        auto keylet = keylet::account(*input.accountID);
+        if (not sharedPtrBackend_->fetchLedgerObject(keylet.key, lgrInfo.seq, ctx.yield))
+            return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+    }
+
+    auto ammKeylet = ammID != 0 ? keylet::amm(ammID) : keylet::amm(input.issue1, input.issue2);
+    auto const ammBlob = sharedPtrBackend_->fetchLedgerObject(ammKeylet.key, lgrInfo.seq, ctx.yield);
+
+    if (not ammBlob)
+        return Error{Status{RippledError::rpcACT_NOT_FOUND, "Amm account not found"}};
+
+    auto const amm = SLE{SerialIter{ammBlob->data(), ammBlob->size()}, ammKeylet.key};
+    auto const ammAccountID = amm.getAccountID(sfAccount);
+    auto const accBlob =
```

Comment:
do we need to check the existence of ammAccountID which is from the amm ledger object.